pkg/test/nodes.go

```diff
@@ -67,15 +67,21 @@ func Node(overrides ...NodeOptions) *v1.Node {
 	}
 }
 
+// this check is done for tests that assume the nodeclaim is past registration but will not be reconciled by the NodeClaim Lifecycle controller
 func NodeClaimLinkedNode(nodeClaim *v1beta1.NodeClaim) *v1.Node {
+	taints := append(nodeClaim.Spec.StartupTaints, nodeClaim.Spec.Taints...)
```

Comment:
nit: Have we thought about keeping the old code, assigning Node to `n` and then appending the taints after creating the node object all in one go